File: demo/MovieousShortVideoDemo/app/src/main/java/video/movieous/media/demo/MyApplication.java
package video.movieous.media.demo;

import android.app.Application;
import android.content.Context;
import android.support.multidex.MultiDex;
import video.movieous.droid.player.MovieousPlayerEnv;
import video.movieous.engine.base.utils.ULog;
import video.movieous.shortvideo.UShortVideoEnv;

/**
 * MyApplication
 */
public class MyApplication extends Application {
    public static MyApplication gContext = null;
    public static final String MOVIEOUS_SHORTVIDEO_SIGN = "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJhcHBpZCI6InZpZGVvLm1vdmllb3VzLm1lZGlhLmRlbW8ifQ.AZGkx11Hojggmn03DXGmFPor6SfSLfOUroDyyycZl5o";
    public static final String MOVIEOUS_PLAYER_SIGN = "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJhcHBpZCI6InZpZGVvLm1vdmllb3VzLm1lZGlhLmRlbW8ifQ.AZGkx11Hojggmn03DXGmFPor6SfSLfOUroDyyycZl5o";

    @Override
    protected void attachBaseContext(Context base) {
        super.attachBaseContext(base);
        MultiDex.install(this);
        gContext = this;
    }

    @Override
    public void onCreate() {
        super.onCreate();

        // 初始化短视频 SDK
        initShortVideoEnv();
        // 初始化播放器 SDK
        initMovieousPlayerEnv();
    }

    private void initShortVideoEnv() {
        UShortVideoEnv.init(this, MOVIEOUS_SHORTVIDEO_SIGN);
        UShortVideoEnv.setLogLevel(ULog.I);
    }

    // 不使用 MovieousPlayer SDK 可以不调用, 具体用法请参考播放器 demo
    private void initMovieousPlayerEnv() {
        // 初始化 SDK，必须第一个调用，否则会出现异常
        MovieousPlayerEnv.init(gContext, MOVIEOUS_PLAYER_SIGN);
    }
}

File: demo/Movieous/app/src/main/java/com/movieous/media/MyApplication.kt
package com.movieous.media

import android.app.Application
import android.content.Context
import com.movieous.media.mvp.model.VideoDataUtil
import com.movieous.media.utils.DisplayManager
import iknow.android.utils.BaseUtils
import video.movieous.droid.player.MovieousPlayerEnv
import video.movieous.engine.base.utils.ULog
import video.movieous.shortvideo.UShortVideoEnv
import kotlin.properties.Delegates

class MyApplication : Application() {

    companion object {
        private val TAG = "MyApplication"
        var context: Context by Delegates.notNull()
    }

    override fun onCreate() {
        super.onCreate()
        context = applicationContext
        DisplayManager.init(this)

        initShortVideoEnv()
        initMovieousPlayerEnv()
        BaseUtils.init(this)
        // 获取播放列表
        VideoDataUtil.doGetVideoList()
    }

    // 初始化短视频 SDK 运行环境
    private fun initShortVideoEnv() {
        UShortVideoEnv.setLogLevel(ULog.I)
        UShortVideoEnv.init(context, Constants.MOVIEOUS_SIGN)
    }

    // 初始化播放器 SDK 运行环境
    private fun initMovieousPlayerEnv() {
        // 初始化 SDK，必须第一个调用，否则会出现异常
        MovieousPlayerEnv.init(context)
        // 开启本地缓存，可以离线播放, 需要 okhttp 支持
        MovieousPlayerEnv.setCacheInfo(cacheDir, null, (100 * 1024 * 1024).toLong(), "MovieousPlayer20", true)

    }

}
